File: app/src/main/java/com/stackanswer/source/local/room/showfavorite/ShowFavoriteDao.java
package com.stackanswer.source.local.room.showfavorite;

import androidx.paging.DataSource;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface ShowFavoriteDao {

    @Query("SELECT * FROM show_table ORDER BY id ASC")
    DataSource.Factory<Integer, ShowFavorite> getAllShow();

    @Insert(onConflict= OnConflictStrategy.REPLACE)
    void insert(ShowFavorite showFavorite);

    @Update
    void update(ShowFavorite showFavorite);

    @Delete
    void delete(ShowFavorite showFavorite);

    @Query("DELETE FROM show_table")
    void deleteAllShow();
}

File: app/src/main/java/com/stackanswer/source/local/room/showfavorite/ShowFavoriteDao.kt
package com.stackanswer.source.local.room.showfavorite

import androidx.room.*
import io.reactivex.Completable
import io.reactivex.Flowable

@Dao
interface ShowFavoriteDao {

    @Query("SELECT * FROM show_table ORDER BY id ASC")
    fun getAllShow(): Flowable<List<ShowFavorite>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(favorite: ShowFavorite): Completable

    @Update
    fun update(favorite: ShowFavorite?)

    @Delete
    fun delete(favorite: ShowFavorite?)

    @Query("DELETE FROM show_table")
    fun deleteAllShow()
}